content/post/building-a-data-practice.md

```diff
@@ -0,0 +1,85 @@
+Start documenting all the places where data is stored, processed, and accessed. Does your organization have multiple databases? Does it have external data sources? Are there systems moving data from one place to another? How are people interacting with data? Write this all down, and include links and information about how to get access.
+
+Eventually this documentation can be part of an internal wiki with illustrations showing a dependency graph, but don't bother with that yet. Just make a Google doc for now.
+
+While mapping the data infrastructure, start thinking about what is missing, what you would like to change, and what the future infrastructure might look like. These ideas can go on your roadmap, which I cover next.
+
+### Plan your roadmap
+One of the perennial Data team struggles is balancing the short term and the long term. You will be busy right away, pulled in multiple directions, but you should take the time to form opinions about the future. How would you like to evolve data tooling, infrastructure, processes, and people?
+
+Think about what is missing, what can be improved, and how they align with the organization's priorities. Are you using Google Analytics for tracking? That might be fine for now, but many businesses will eventually need something more powerful. How are reports being run? If data is primarily accessed and explored in spreadsheets, you should think about options for BI tools. Who should be hired next? A data analyst? A data engineer?
+
+Form opinions about your roadmap now so you will be prepared to advocate for it whenever it comes up. What will you say when someone in marketing decides they want to integrate a new service, or when an engineer starts to build a data tool in house, or when the founders raise more funding and ask you how to spend it?
+
+### Create a data dictionary
+As you learn things, especially things that are surprising, write them down. Think about what the next data hire will need to know. Avoid the urge to make this fancy — a Google doc is fine.
+
+The concept of a data dictionary is a bit controversial. What do you include? What don't you include? How do you maintain it? For more mature organizations, I recommend considering [Carl Anderson's suggestions](https://medium.com/@leapingllamas/data-dictionary-a-how-to-and-best-practices-a09a685dcd61) for using a data dictionary to create a shared understanding around key metrics and vocabulary.
+
+In this case, though, a data dictionary is primarily for you and future data hires. Even though the business and its data will change over time, there are still key pieces of information that are useful to document. Are timezones in UTC or your organization's time? What are some core pieces of business logic? How are key metrics calculated? Are there important database tables or column names that are not self-explanatory?
+
+At Kickstarter, we called this document *Common Data Gotchas*, which is a good way to think about it.
```

Comment:
I really like this -- I think it's key to _not_ get hung up on making it exhaustive (a list of every column should be an explicit anti-goal). You could maybe drive that home with something like:

> Do _not_ set yourself the goal of creating an exhaustive data dictionary. The completionists find it challenging to leave all of those columns undocumented, but not only will it be time consuming to create, a pain to manage, and seldom read, but despite your best efforts it will get quickly out of date.

Reply:
Agreed. I would add that even before describing columns, you can just get alignment and clarity around definitions of key concepts. For example, when we (at Trialspark) talk about "active studies", different teams have different use cases and definitions. It's helpful to align on the concept first ("active recruitment studies" "active FDA studies" etc...) and then figure out where and how that maps to the columns.